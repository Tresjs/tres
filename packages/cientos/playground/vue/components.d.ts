/* eslint-disable */
// @ts-nocheck
// Generated by unplugin-vue-components
// Read more: https://github.com/vuejs/core/pull/3399
// biome-ignore lint: disable
export {}

/* prettier-ignore */
declare module 'vue' {
  export interface GlobalComponents {
    AkuAku: typeof import('./src/components/gltf/BlenderCube.vue')['default']
    BlenderCube: typeof import('./src/components/gltf/BlenderCube.vue')['default']
    FboCube: typeof import('./src/components/FboCube.vue')['default']
    Gltf: typeof import('./src/components/gltf/index.vue')['default']
    GraphPane: typeof import('./src/components/GraphPane.vue')['default']
    Jeep: typeof import('./src/components/fbx/Jeep.vue')['default']
    ModelsDemo: typeof import('./src/components/ModelsDemo.vue')['default']
    OverlayInfo: typeof import('./src/components/OverlayInfo.vue')['default']
    RouterLink: typeof import('vue-router')['RouterLink']
    RouterView: typeof import('vue-router')['RouterView']
    StarsForTest: typeof import('./src/components/StarsForTest.vue')['default']
  }
}
