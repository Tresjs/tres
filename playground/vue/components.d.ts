/* eslint-disable */
// @ts-nocheck
// Generated by unplugin-vue-components
// Read more: https://github.com/vuejs/core/pull/3399
export {}

/* prettier-ignore */
declare module 'vue' {
  export interface GlobalComponents {
    AkuAku: typeof import('./src/components/AkuAku.vue')['default']
    AnimatedObjectUseUpdate: typeof import('./src/components/AnimatedObjectUseUpdate.vue')['default']
    BlenderCube: typeof import('./src/components/BlenderCube.vue')['default']
    Box: typeof import('./src/components/Box.vue')['default']
    DirectiveSubComponent: typeof import('./src/components/DirectiveSubComponent.vue')['default']
    DynamicModel: typeof import('./src/components/DynamicModel.vue')['default']
    FBOCube: typeof import('./src/components/FBOCube.vue')['default']
    GraphPane: typeof import('./src/components/GraphPane.vue')['default']
    LocalOrbitControls: typeof import('./src/components/LocalOrbitControls.vue')['default']
    Overlay: typeof import('./src/components/Overlay.vue')['default']
    OverlayInfo: typeof import('./src/components/OverlayInfo.vue')['default']
    PbrSphere: typeof import('./src/components/PbrSphere.vue')['default']
    RouterLink: typeof import('vue-router')['RouterLink']
    RouterView: typeof import('vue-router')['RouterView']
    TakeOverLoopExperience: typeof import('./src/components/TakeOverLoopExperience.vue')['default']
    TestResult: typeof import('./src/components/TestResult.vue')['default']
    Tests: typeof import('./src/components/Tests.vue')['default']
    TestSphere: typeof import('./src/components/TestSphere.vue')['default']
    Text3D: typeof import('./src/components/Text3D.vue')['default']
    TheCameraOperator: typeof import('./src/components/TheCameraOperator.vue')['default']
    TheExperience: typeof import('./src/components/TheExperience.vue')['default']
    TheSphere: typeof import('./src/components/TheSphere.vue')['default']
    UgglyBunny: typeof import('./src/components/UgglyBunny.vue')['default']
  }
}
