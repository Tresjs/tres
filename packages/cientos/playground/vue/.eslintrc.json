{
  "extends": "@tresjs/eslint-config-vue",
  "rules": {
    "no-console": "off",
    "vue/attribute-hyphenation": [2, "always", {
      "ignore": [
        "shadow-mapSize-width",
        "shadow-mapSize-height",
        "distortionMap",
        "material-uniforms-mieCoefficient-value",
        "material-uniforms-mieDirectionalG-value",
        "material-uniforms-sunPosition-value"
      ]
    }]
  }
}
