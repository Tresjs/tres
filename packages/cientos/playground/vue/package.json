{
  "name": "playground",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite --host",
    "build": "vue-tsc && vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@tresjs/core": "5.0.0",
    "vue-router": "^4.5.1"
  },
  "devDependencies": {
    "@tresjs/leches": "^0.14.1",
    "unplugin-auto-import": "^20.1.0",
    "unplugin-vue-components": "^29.0.0",
    "vite-plugin-glsl": "^1.5.1",
    "vite-plugin-qrcode": "^0.3.0",
    "vue-tsc": "^3.0.7"
  }
}
