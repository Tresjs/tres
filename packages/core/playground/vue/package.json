{
  "name": "playground",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite --host",
    "build": "vite build",
    "preview": "vite preview" 
  },
  "dependencies": {
    "@tresjs/cientos": "5.0.0",
    "@tresjs/core": "workspace:^",
    "@tresjs/leches": "https://pkg.pr.new/@tresjs/leches@b34e795",
    "vue-router": "^4.5.1"
  },
  "devDependencies": {
    "@tweakpane/plugin-essentials": "^0.2.1",
    "unplugin-auto-import": "^20.2.0",
    "vite-plugin-glsl": "^1.5.1",
    "vite-plugin-qrcode": "^0.3.0",
    "vite-plugin-vue-devtools": "8.0.2",
    "vue-tsc": "^3.0.8"
  }
}
