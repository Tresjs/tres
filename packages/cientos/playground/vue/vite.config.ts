import { templateCompilerOptions } from '@tresjs/core'
import vue from '@vitejs/plugin-vue'
import { resolve } from 'pathe'
import UnoCSS from 'unocss/vite'
import AutoImport from 'unplugin-auto-import/vite'
import Components from 'unplugin-vue-components/vite'
import { defineConfig } from 'vite'
import glsl from 'vite-plugin-glsl'
import { qrcode } from 'vite-plugin-qrcode'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    glsl(),
    vue(templateCompilerOptions),
    AutoImport({
      dts: true,
      eslintrc: {
        enabled: true, // <-- this
      },
      imports: ['vue'],
    }),
    Components({
      /* options */
    }),
    UnoCSS({
      theme: {
        colors: {
          'cientos-blue': '#82dbc5',
          'cientos-orange': '#fbb03b',
        },
      },
    }),
    qrcode(),
  ],
  resolve: {
    alias: {
      '@tresjs/cientos': resolve(__dirname, '../../src/'),
    },
  },
})
