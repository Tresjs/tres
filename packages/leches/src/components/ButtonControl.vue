<script setup lang="ts">
import { computed } from 'vue'
import type { Control } from '../types'

const props = defineProps<{
  label: string
  control: Control
}>()

const classes = computed(() => {
  const base = 'leches-btn'
  const variant = `leches-btn-${props.control.value.variant || 'primary'}`
  const size = `leches-btn-${props.control.value.size || 'sm'}`
  return [base, variant, size]
})
</script>

<template>
  <button
    :class="classes"
    @click="control.value.onClick"
  >
    <i
      :class="[control.value.icon, control.value.label ? 'mr-1' : '']"
    ></i>
    {{ control.value.label }}
  </button>
</template>

<style scoped>
.leches-btn {
  @apply tl-rounded-md tl-cursor-pointer tl-border-none tl-flex tl-items-center;
  @apply tl-font-bold;
  @apply tl-transition-all tl-duration-200;
}

.leches-btn-sm {
  @apply tl-text-xs tl-px-2.5 tl-py-2;
}

.leches-btn-md {
  @apply tl-text-sm tl-px-3 tl-py-2;
}

.leches-btn-lg {
  @apply tl-text-base tl-px-4 tl-py-2;
}

.leches-btn-block {
  @apply tl-px-4 tl-py-2 tl-text-base tl-w-full tl-justify-center;
}

.leches-btn-primary {
  @apply tl-bg-gray-100 tl-text-gray-700;
}
.leches-btn-primary:hover {
  @apply tl-bg-gray-200;
}

.leches-btn-secondary {
  @apply tl-bg-dark-700 tl-text-light-100;
}

.leches-btn-secondary:hover {
  @apply tl-bg-dark-600;
}
</style>
