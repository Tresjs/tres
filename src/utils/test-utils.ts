import { createApp } from 'vue'
import type { Fn } from '@vueuse/core'

export function withSetup(composable: Fn) {
  let result
  const app = createApp({
    setup() {
      result = composable()
      // suppress missing template warning
      return () => null
    },
  })
  app.mount(document.createElement('div'))
  // return the result and the app instance
  // for testing provide / unmount
  return [result, app]
}
