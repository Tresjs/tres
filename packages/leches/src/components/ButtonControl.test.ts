import { mount } from '@vue/test-utils'
import { TresLeches, useControls } from '/@/'
import { describe, expect, it } from 'vitest'
import { defineComponent, ref } from 'vue'
import ButtonControl from './ButtonControl.vue'

describe('button Controls', () => {
  let wrapper
  let component
  const mountComponent = (setup) => {
    component = defineComponent({
      template: `<TresLeches />`,
      components: { TresLeches, ButtonControl },
      setup,
    })

    wrapper = mount(component, {
      components: { TresLeches, ButtonControl },
    })
  }

  it('should render a button', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').exists()).toBe(true)
  })

  it('should render a button with a label', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').text()).toBe('Accept')
  })

  it('should render a clickable button', async () => {
    mountComponent(() => {
      const isClicked = ref(false)
      useControls({ acceptBtn: { type: 'button', label: 'Accept', onClick: () => { isClicked.value = true } } })
      return { isClicked }
    })

    const button = wrapper.find('button')
    await button.trigger('click')
    expect(wrapper.vm.isClicked).toBe(true)
  })

  it('should render an icon on the button', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept', icon: 'i-carbon-checkmark' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('i.i-carbon-checkmark').exists()).toBe(true)
  })

  it('should render a small button', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept', size: 'sm' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').classes()).toContain('leches-btn-sm')
  })

  it('should render a large button', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept', size: 'lg' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').classes()).toContain('leches-btn-lg')
  })

  it('should render a block size button', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept', size: 'block' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').classes()).toContain('leches-btn-block')
  })

  it('should render a primary button variant by default', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').classes()).toContain('leches-btn-primary')
  })

  it('should render a secondary button variant', () => {
    mountComponent(() => {
      useControls({ acceptBtn: { type: 'button', label: 'Accept', variant: 'secondary' } })
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('button').classes()).toContain('leches-btn-secondary')
  })
})
