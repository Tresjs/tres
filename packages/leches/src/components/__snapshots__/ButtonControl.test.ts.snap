// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Controls > should render a block size button 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-block\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a button 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a button with a label 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a large button 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-lg\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a primary button variant by default 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a secondary button variant 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-secondary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render a small button 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`Button Controls > should render an icon on the button 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">🍰</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class=\\"tl-p-2 tl-inline-flex\\"><button data-v-9a5f9367=\\"\\" class=\\"leches-btn leches-btn-primary leches-btn-sm\\"><i data-v-9a5f9367=\\"\\" class=\\"i-carbon-checkmark mr-1\\"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a block size button 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-block"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a button 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-sm"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a button with a label 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-sm"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a large button 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-lg"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a primary button variant by default 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-sm"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a secondary button variant 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-secondary leches-btn-sm"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render a small button 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-sm"><i data-v-9a5f9367="" class="mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;

exports[`button Controls > should render an icon on the button 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">🍰</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <!--v-if-->
    <div>
      <div class="tl-p-2 tl-inline-flex"><button data-v-9a5f9367="" class="leches-btn leches-btn-primary leches-btn-sm"><i data-v-9a5f9367="" class="i-carbon-checkmark mr-1"></i> Accept</button></div>
    </div>
  </div>
</div>"
`;
